File: src/algorithms/sorting/js/selection-sort.js
export default function selectionSort(originalArray) {

  let minIndex;
  // 원본 보존
  const arr = [...originalArray];

  for (let i = 0; i < arr.length - 1; i++) {
    minIndex = i;

    for (let j = i + 1; j < arr.length; j++) {
      if (arr[minIndex] > arr[j])
        minIndex = j;
    }

    [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
  }

  return arr;
}

const arr = [];
for (let i = 0; i < 15; i++) {
  arr.push(Math.floor(Math.random() * 100));
}

console.log("UnSorted array: ");
console.log(arr);

const sorted = selectionSort(arr);
console.log("Sorted array: ");
console.log(sorted);

File: src/algorithms/sorting/selectionsort.py
# 변수, 함수, 속성 => 스네이크 케이스
# 클래스, 예외 => 파스칼 케이스

num_list = [1,3,4,2,7,8,9,11,5]
def selection_sort(a):

  length = len(a)
  min_index = 0
  for i in range(length-1):
    min_index = i
    for j in range(i+1,length):
      if a[min_index] > a[j]:
        min_index = j
    a[i] , a[min_index] = a[min_index] , a[i]

  return a

print(selection_sort(num_list))



num_list1 = [1,3,4,2,7,8,9,11,5]
def selection_sort(a):

  length = len(a)
  
  for i in range(length-1):
    
    for j in range(i+1,length):
      if a[i] > a[j]:
        a[i] , a[j] = a[j] , a[i]

  return a

print(selection_sort(num_list1))